src/entity_extraction/training/spacy_ner/spacy_preprocess.py

```diff
@@ -0,0 +1,266 @@
+# Author: Jenit Jain
+# Date: 2023-06-21
+"""This script manages the custom training and fine tuning of spacy models.
+
+Usage: spacy_train.py --data_path=<data_path> --train_split=<train_split> --val_split=<val_split> --test_split=<test_split>
+
+Options:
+    --data_path=<data_path>         The path to the evaluation data in json format.
+    --train_split=<train_split>     The ratio of files to include in the training set.
+    --val_split=<val_split>         The ratio of files to include in the validation set.
+    --test_split=<test_split>       The ratio of files to include in the testing set.
+"""
+
+import os, sys
+
+import pandas as pd
+import numpy as np
+import spacy
+import json
+import logging
+import shutil
+import glob
+from docopt import docopt
+from spacy.tokens import DocBin
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+opt = docopt(__doc__)
+
+def preprocess_data():
+    # If the data_path consists of JSON, load all the files in the directory
+    nlp = spacy.blank("en")
+    # Creating artifacts for only train and val sets
+    train_doc_bin, val_doc_bin = DocBin(), DocBin()
+    
+    # Segragating all files for evaluation purposes
+    train_data = {}
+    val_data = {}
+    test_data = {}
+    
+    train_path = os.path.join(opt['--data_path'], "train")
+    val_path = os.path.join(opt['--data_path'], "val")
+    test_path = os.path.join(opt['--data_path'], "test")
+
+    if os.path.exists(train_path):
+        shutil.rmtree(train_path)
+        logger.info(f"The folder '{train_path}' has been deleted.")
+    if os.path.exists(val_path):
+        shutil.rmtree(val_path)
+        logger.info(f"The folder '{val_path}' has been deleted.")
+    if os.path.exists(test_path):
```

Comment:
issue: this is duplicating the process done in the `labelling_data_split.py` file, instead run that script first in bash with these args then make this function run with those processed folders, that way we remove code duplication and make the parts of the process more re-usable.